gdpr/impl.go

```diff
@@ -181,17 +180,7 @@ func (p *permissionsImpl) allowPITCF2(parsedConsent api.VendorConsents, vendor a
 			break
 		}
 	}
-	// Set to true so any purpose check can flip it to false
-	allowPI = true
-	if p.cfg.TCF2.Purpose1.Enabled {
-		allowPI = allowPI && p.checkPurpose(consent, vendor, vendorID, consentconstants.InfoStorageAccess, weakVendorEnforcement)
-	}
-	if p.cfg.TCF2.Purpose2.Enabled {
-		allowPI = allowPI && p.checkPurpose(consent, vendor, vendorID, consentconstants.BasicAdserving, weakVendorEnforcement)
-	}
-	if p.cfg.TCF2.Purpose7.Enabled {
-		allowPI = allowPI && p.checkPurpose(consent, vendor, vendorID, consentconstants.AdPerformance, weakVendorEnforcement)
-	}
```

Comment:
I think it is outdated as I don't see this logic anywhere in the TCF 2.0 spec. Either that or we have a bug because the `allowPI` value isn't being used anywhere 🙂.  The purpose enabled config flags will be added in the near future in separate PRs to appropriately set the allowID and allowGeo flags.

Reply:
This logic was added as part of TCF 2.0 about a year ago. I'm wondering if the requirements were misunderstood or have changed since this logic was added. `allowPI` was used to determine whether user IDs should be stripped when TCF 2.0 was first implemented and this logic is saying to strip it if at least one of the enabled purposes does not have consent while the requirements say that they should be stripped if at least one of the enabled purposes has consent.

From the spec:
> 19) User IDs must be stripped from a given vendor’s auction in these scenarios:
- User has not provided a legal basis for any Purpose 2,3,4,5,6,7,8,9, or 10 and the publisher has not defined an exception for the vendor for any of these purposes.

Also from the spec:
**Activity** - Passing User IDs into an auction
**Legal Basis Required** - Any Purpose 2-10. User IDs are important for more than personalizing ads - they can be used in frequency capping, building profiles, counting unique users, etc. So Prebid Server should pass User IDs through the auction if any of Purposes 2-10 pass the legal basis test.

I was planning to add this logic under a separate PR.

@hhhjort thoughts?

Reply:
We have both AllowID and AllowPI. So lines 177-182 take care of AllowID for point 19 from the spec quoted above. AllowPI is to flag other non-geo personal information, like age or gender, for screening or not. The valid uses cases for these other pieces of info were considered more limited, and thus only available for a limited selection of purposes.